### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by matteosz**
This PR increase the test coverage for the modules util/encoding and util/encryption

### CODE REVIEW

File: util/encoding/encoding_test.go [hunk 1]
```diff
@@ -60,3 +61,38 @@ func TestScalarHexString(t *testing.T) {
...
+	mockReader1 := &MockFailingReader{data: []byte("abc")}
+	_, err = ReadHexPoint(s, mockReader1)
+	require.Error(t, err, "Expected error when reader fails, but got nil")
+
+	// Test case: not enough bytes from stream
+	mockReader2 := &MockEmptyReader{data: []byte("abc")}
+	_, err = ReadHexPoint(s, mockReader2)
+	require.Error(t, err, "Expected error when not enough bytes from stream, but got nil")
+	if err.Error() != "didn't get enough bytes from stream" {
+		t.Errorf("Expected error message: didn't get enough bytes from stream, but got %s", err.Error())
```
Comment (AnomalRoil) [inline util/encoding/encoding_test.go:96]: This could be done using `require.ErrorContains`.

**Review by AnomalRoil — APPROVED** [review]
LGTM, just one comment.

File: util/encoding/encoding_test.go (hunk 1 above)
Comment (matteosz) [reply util/encoding/encoding_test.go]: I used `require.EqualError` to exactly match it

**Review by AnomalRoil — DISMISSED** [review]

File: util/random/rand_test.go [hunk 2]
```diff
@@ -84,3 +85,59 @@ func TestIncorrectSize(t *testing.T) {
...
+		bitlen uint // input bit length
+		exact  bool // whether the exact bit length should be enforced
+	}{
+		{bitlen: 128, exact: false},
+		{bitlen: 256, exact: true},
+		{bitlen: 512, exact: false},
+		{bitlen: 1024, exact: true},
+	}
+
+	for _, tc := range testCases {
```
Comment (pierluca) [inline util/random/rand_test.go:100]: Nice pattern, but to improve test results reporting you'll want to use the Go sub-test pattern:
https://go.dev/blog/subtests

```
    for _, tc := range testCases {
        t.Run(fmt.Sprintf("bitlen: %d exact: %s", tc.bitlen, tc.exact), func(t *testing.T) {
            ...
        })
    }
```

same obviously applies to the other test :)

File: util/encoding/encoding_test.go [hunk 3]
```diff
@@ -60,3 +61,36 @@ func TestScalarHexString(t *testing.T) {
...
+}
+
+func (m *MockFailingReader) Read(p []byte) (n int, err error) {
+	return copy(p, m.data), io.EOF
+}
+func (m *MockEmptyReader) Read(p []byte) (n int, err error) {
+	return 0, nil
+}
+
+func TestReadHexPointError(t *testing.T) {
```
Comment (pierluca) [inline util/encoding/encoding_test.go:80]: I'm going to be annoying on a point that's not common in this codebase, that is a bit of a pet peeve of mine, but which I think is important nonetheless:

This test contains 3 different test cases. When possible, try to have independent unit tests.
- This way, when the test fails, it tells you directly that it's the "reader fails" that fails, you don't have to look up which line this failed at.
- Furthermore, if the 3rd test case passes, you won't be able to tell (from the test results) if everything's grouped in a single test.
- Finally, having separate tests makes them more (trivially) independent of one another, and potentially parallelizable (not that it matters here)

**Review by pierluca — COMMENTED** [review]
Good increment, I've suggested some improvements :)

**Review by pierluca — APPROVED** [review]

**Review by AnomalRoil — APPROVED** [review]

**Pull request merged**